sdk/identity/azure-identity/inc/azure/identity/azure_pipelines_credential.hpp

```diff
@@ -58,7 +58,7 @@ namespace Azure { namespace Identity {
   private:
     std::string m_serviceConnectionId;
     std::string m_systemAccessToken;
-    Azure::Core::Http::_internal::HttpPipeline m_httpPipeline;
+    std::unique_ptr<Azure::Core::Http::_internal::HttpPipeline> m_httpPipeline;
```

Comment:
Do you want `unique_ptr` or `shared_ptr` here? `unique_ptr` means that a pipeline credential object cannot be copied, which may or may not be what you desire.

From what I've seen in the past, most folks seem to use `shared_ptr` for their pipelines. But for a credential, it may make sense to use a unique pointer since credentials are usually accessed via `shared_ptr` objects.

Reply:
Without looking much into it, I'd say if we want the credential to be copyable, it would be a safer bet to still use the unique_ptr to a pipeline, but implement copy constructor and assignment, which would clone the pipeline.
Otherwise, when used as shared, I am not really sure how much state is shared (policies, such as BearerTokenAuthPolicy, do have state - they cache the token) and if that is necessary/if it would be a correct behavior.

Reply:
We want it to be `unique_ptr` here. I don't see a reason for it to be a `shared_ptr`. All the other credentials already have the HTTP pipeline as a `unique_ptr` to the TokenCredentailImpl, so it can't be copied:
https://github.com/Azure/azure-sdk-for-cpp/blob/7fea3ff67756eb0fcd5af6cc5ed34411eb888403/sdk/identity/azure-identity/src/private/client_assertion_credential_impl.hpp#L25